### PR TITLE
YAML fixes

### PR DESCRIPTION
**Pull request by glenn-jocher**
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Docs, examples, and configs were polished for clarity and safety, with example guides updated to reference YOLO11 and a few small robustness fixes across tooling. ✅✨

### 📊 Key Changes
- 📚 **Documentation cleanup & consistency**
  - Fixed grammar, capitalization (e.g., “iOS”), punctuation, and improved link formatting (e.g., Codecov + Contributor Covenant links).
  - Renamed/standardized headings (e.g., “Minimum Reproducible Example”) and refined metadata descriptions/keywords.
- 🧪 **Test callback compatibility**
  - Updated `tests/test_engine.py` callback signature to accept `**kwargs`, improving compatibility with callback invocations that pass keyword args.
- 🐳 **Jetson Docker reliability**
  - Made `pip` symlink creation forced (`ln -sf`) to prevent failures when the link already exists.
- 🦀 **Rust ONNXRuntime example hardening**
  - Added a **10MB cap** when downloading font files to avoid unbounded reads.
  - Prevented divide-by-zero in IoU calculation when union area is zero.
- 🖥️ **OpenVINO C++ example cleanup**
  - Added `cv::destroyAllWindows()` after displaying results to properly close OpenCV windows.
- 🧩 **Model/config comment fixes**
  - Corrected misleading `scales` comments in RT-DETR/YOLOv6/YOLOv10 YAMLs and minor typos (e.g., “yolov3-tiiny” → “yolov3-tiny”).
- 🧭 **Example README updates to YOLO11**
  - Updated Action Recognition and Region Counter READMEs to reference **YOLO11** weights (`yolo11n.pt`, `yolo11m.pt`) and adjusted a couple defaults (e.g., `--skip-frame`).

### 🎯 Purpose & Impact
- 🧠 **Clearer, more professional docs** → easier onboarding and fewer user misunderstandings when following guides.
- 🛡️ **More robust examples** (Rust/OpenVINO) → fewer crashes/hangs and safer handling of external downloads.
- 🧪 **Fewer CI/test edge cases** → callbacks won’t break if the engine passes extra keyword arguments.
- 🐳 **Smoother Jetson builds** → reduces Docker build failures due to existing symlinks.
- 🧾 **More accurate model config guidance** → reduces confusion when editing or scaling model YAMLs.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message—an engineer will review and assist shortly 🤖✅ Please review the checklist below to help ensure a smooth merge:

-✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
-✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
-✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
-✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
-✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
-✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
-✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: examples/YOLOv8-ONNXRuntime-Rust/src/lib.rs [hunk 1]
```diff
@@ -96,15 +96,7 @@ pub fn check_font(font: &str) -> rusttype::Font<'static> {
...
-        let total_size = resp
-            .header("Content-Length")
-            .and_then(|s| s.parse::<u64>().ok())
-            .unwrap();
-        let _reader = resp
-            .into_reader()
-            .take(total_size)
-            .read_to_end(&mut buffer)
-            .unwrap();
+        resp.into_reader().read_to_end(&mut buffer).unwrap();
```
Comment (UltralyticsAssistant) [inline examples/YOLOv8-ONNXRuntime-Rust/src/lib.rs]: ⚠️ **HIGH**: Removing the `Content-Length` bound changes semantics: `read_to_end()` will read until EOF and can buffer an arbitrarily large response if the server misbehaves or a redirect serves unexpected content. This can cause excessive memory usage or a hang. Consider keeping a hard cap (even without relying on `Content-Length`) and failing fast when exceeded.

File: examples/YOLOv8-ONNXRuntime-Rust/src/lib.rs [hunk 2]
```diff
@@ -139,15 +131,7 @@ pub fn load_font() -> FontArc {
...
-            let total_size = resp
-                .header("Content-Length")
-                .and_then(|s| s.parse::<u64>().ok())
-                .unwrap();
-            let _reader = resp
-                .into_reader()
-                .take(total_size)
-                .read_to_end(&mut buffer)
-                .unwrap();
+            resp.into_reader().read_to_end(&mut buffer).unwrap();
```
Comment (UltralyticsAssistant) [inline examples/YOLOv8-ONNXRuntime-Rust/src/lib.rs]: ⚠️ **HIGH**: Same issue as above: `read_to_end()` without any upper bound can lead to unbounded memory growth when downloading the font. Add a maximum allowed size (e.g. a few MB) and abort if exceeded to make the example safer and more robust.

File: ultralytics/cfg/models/rt-detr/rtdetr-resnet101.yaml [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 # Parameters
 nc: 80 # number of classes
-scales: # model compound scaling constants, i.e. 'model=yolov8n-cls.yaml' will call yolov8-cls.yaml with scale 'n'
+scales: # model compound scaling constants, i.e. 'model=rtdetr-resnet101.yaml' will call rtdetr-resnet101.yaml with scale 'l'
```
Comment (UltralyticsAssistant) [inline ultralytics/cfg/models/rt-detr/rtdetr-resnet101.yaml:9]: 💡 **MEDIUM**: The comment says the config uses scale `'l'`, but these RT-DETR ResNet configs typically represent a single scale variant and the example should match the key(s) defined below. If this file only defines `l: [...]`, it's OK, but the phrasing is confusing because `model=rtdetr-resnet101.yaml` already implies the variant. Consider aligning the comment with how scales are actually selected in Ultralytics (e.g., “uses the `l` scale”) to avoid implying a self-referential call pattern.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Overall looks clean: docs/link anchor tweaks, YOLO11 naming updates, test callback accepting `**kwargs`, and the OpenVINO window cleanup are all reasonable. The main concern is the Rust example now using unbounded `read_to_end()` for network downloads, which can lead to excessive memory use/hangs; adding a max-size guard would make it safer. A couple RT-DETR `scales` comment updates read a bit self-referential/misleading and could be clarified to reflect how scale selection actually works.

💬 Posted 4 inline comments

File: ultralytics/cfg/models/rt-detr/rtdetr-resnet50.yaml [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
 # Parameters
 nc: 80 # number of classes
-scales: # model compound scaling constants, i.e. 'model=yolov8n-cls.yaml' will call yolov8-cls.yaml with scale 'n'
+scales: # model compound scaling constants, i.e. 'model=rtdetr-resnet50.yaml' will call rtdetr-resnet50.yaml with scale 'l'
```
Comment (UltralyticsAssistant) [inline ultralytics/cfg/models/rt-detr/rtdetr-resnet50.yaml:9]: 💡 **MEDIUM**: Same as rtdetr-resnet101.yaml: the updated `scales` comment is potentially misleading about how the `l` scale is selected. Suggest adjusting to “uses the `l` scale” (or similar) to match actual selection behavior and avoid confusion for users editing configs.

**Comment by UltralyticsAssistant** [comment]
Merged—huge thanks for this, @glenn-jocher. Your steady polish across docs, examples, configs, and tooling makes the whole project feel safer, clearer, and more welcoming to everyone who lands here next.

> “If I have seen further it is by standing on the shoulders of giants.” — Isaac Newton

This PR is exactly that kind of foundational work: sharper documentation and consistent YOLO11 references improve onboarding, while the callback, Jetson Docker, Rust ONNXRuntime, and OpenVINO hardening quietly removes friction and failure modes for countless users. Appreciate the care and craftsmanship—these “small” fixes compound into a much better developer experience.
